### DISCUSSION TITLE
Issue setting up POSClient with ethers.js and getting erc20 balance

### DISCUSSION THREAD

**Discussion by tylerpashigian** [0xPolygon/matic.js#355] [Q&A]
I am trying to use the POSClient for the first time and set up a few dev wallets while walking through [this tutorial](https://medium.com/the-polygon-blog/beginner-friendly-tutorial-to-matic-js-58f7c24c9744) combined with the [Polygon docs](https://wiki.polygon.technology/docs/develop/ethereum-polygon/matic-js/pos/) and [matic.js examples](https://github.com/maticnetwork/matic.js/tree/master/examples) and I believe I have set up everything for using the Mumbai testnet.

I am trying to just do a simple erc20.getBalance() of an address of one of the addresses I loaded dummy erc20 on via the mumbai faucet, but I am getting the following error: `erc20.getBalance()): Error: could not detect network (event="noNetwork", code=NETWORK_ERROR, version=providers/5.7.0)`. I ended up creating a contract and provider using raw ethers.js and successfully read the balance, but would really like to use the matic.js client if possible. I assume the rpc's I was using are the issue since the address and token contract seem to be working fine. I will provide the network info I was using below to see if anyone can see a problem. Please let me know if anyone has ideas, thanks!

**Network Info In .env File** - [Network Info Source](https://docs.polygon.technology/docs/develop/network-details/network/#polygon-pos-testnet)
`ROOT_RPC = https://rpc.goerli.mudit.blog`
`MATIC_RPC = https://rpc-mumbai.matic.today`

**Comment by ujjwalgupta94** [comment]
Could you provide me a reproducable repo please ? @tylerpashigian

**Reply by tylerpashigian** [reply]
I am hosting mine from within a firebase function project, here is the repo: https://github.com/tylerpashigian/march-madness

USER1_FROM='0x1f11969BB916Cd4af929B44B5F24Dd5162bfFABd'
USER1_PRIVATE_KEY=${PRIVATE_KEY}
ROOT_RPC='https://rpc.goerli.mudit.blog'
MATIC_RPC='https://rpc-mumbai.matic.today'
